### ISSUE TITLE
Layout Save Triggering

### ISSUE THREAD

**Issue by McCoyCOREUSA** [smeighan/xLights#3648]
v2022.25

Any changes to only sequences triggers pop-up to save layout/model changes. Happens frequently and I never even accessed the controllers or layout tabs.

Inquired with Zoom Room and many others report the same issue. Some reported this was more random in .24, too.

**Comment by computergeek1507** [comment]
If you edit the display elements, the views are also saved to the layout files so they can be reused between sequences.

**Comment by McCoyCOREUSA** [comment]
Sorry, I'm not sure I follow?

I opened xLights.
I opened a sequence.
I adjusted an ON effect by dragging the ends to make it a bit shorter.
I saved, rendered, saved.
FPP Connect uploaded the sequence.
Clicked the close to close xLights and I get a pop-up that models/layout changed and asked if I wanted to save those changes.

Which step triggered this change to the layouts/models?

Thanks,
Scott

**Comment by McCoyCOREUSA** [comment]
Scott, does this mean I need to re-push my outputs every time to make sure it's all matching?

**Comment by computergeek1507** [comment]
Unless some setting magically changes between the last app shutdown and startup, the start channels should not change. If it was a major issue I think more people would have noticed by now.

**Comment by McCoyCOREUSA** [comment]
I agree. I noticed it in .24, but just re-pushed everything. I thought I was doing something to cause it. I asked the Zoom Room last night and most everyone said they had the same issue. So I said I'd file a bug report. Thanks.

**Comment by computergeek1507** [comment]
It's been a bug for over a year, it is also the reason you have to rerender sequences after opening them after restarting xLights. It also sets a flag that the sequence buffer size has changed and the previous FSEQ data is "old" and incorrect

**Comment by AzGilrock** [comment]
The trigger to save the layout has been an on and off thing for years.  I've gotten used to just keeping in mind whether I've done anything that needs saving.  There are so many spots that flag the save it's hard to track down.  Well it might be easier now that I've gotten better at debuggin on the mac.

**Comment by merryoncherry** [comment]
I've noticed that the button for "save" on the layout tab is simply red when it shouldn't be.  If you press it, it stays red.  If you switch to controller tab and back, it clears.  One on controller tab works properly, it clears after you press save.  (I do have them linked.)  There can be more than one issue, of course, but I wonder if this is part of it.

**Comment by McCoyCOREUSA** [comment]
Merryoncherry - that has also been reported in the Zoom Room a few times. We found that clicking the save button on the controllers tab worked when the save button in the layout tab stays red when clicked.

**Comment by keithsw1111** [comment]
@merryoncherry is this closed with your change?

**Comment by merryoncherry** [comment]
I think there could be a different issue here, with xLights thinking it needs to save when it shouldn't think that.  Or multiple other issues.

All I did was fix one of the visual indicators that a save needed to happen, when the save had already happened.

**Comment by computergeek1507** [comment]
I have seen this too, when xLights loads the network file, the start channels are recalculated. It causes the dirty flag to be set true. It also invalidates the channel buffer, requiring users to rerender sequences on open.

**Comment by keithsw1111** [comment]
Do we have a sample show folder that reliably does it every time it opens ... I can start to run down the causes then

**Comment by merryoncherry** [comment]
I feel bad even suggesting it... but my rookie year 2021 show folder does it, just tested in 2023.02.  It does have some "check sequence" issues with missing images, maybe that's the first thing to eliminate to see if it's related, otherwise it's something else buried in there somewhere...

https://drive.google.com/file/d/1h4DoeB96RcGQBT7ZMqv91ZkoBLTo5wpJ/view?usp=share_link

**Comment by keithsw1111** [comment]
So this is happening because you are running a hinkspix ... with universe per string. The code in ReworkStartChannels redoes all the start channels and when this option is set determining if anything changed becomes so complex I just gave up and said it always changed. I realise that is not optional but this is seriously messy to handle.

**Comment by merryoncherry** [comment]
Thanks for the explanation.  I will take a quick look in there and see if it looks like it could be done, or better left alone.

**Comment by cybercop23** [comment]
So besides misconfigured HPs with UPS, sometimes you'll still get the Save flag, but the expectation is not nothing changes.
Things get flagged "dirty" too often, but unless you made changes in your layout, then channels will not change.

Should this continue to stay open?

**Comment by derwin12** [comment]
This is likely a duplicate of the HP universe per string issue that has a PR for on another ticket.
